File: code/ruby/app/controllers/home_controller.rb
class HomeController < ApplicationController
  def index
    @games = Array.new
    GameBase::AVAILABLE_GAME_TYPES.each {|type|
      game_summary = GameSummary.new
      if (type == 'DrawFiveGame')
        game_summary.name = 'Draw Five'

        game = DrawFiveGame.new
        high_scores = game.get_high_scores
        highest_score = high_scores.length > 0 ? high_scores[0] : nil

        if (highest_score != nil)
          game_summary.high_score_name = highest_score.name
          game_summary.high_score = highest_score.score
        end
        game_summary.link = '/Draw_Five/index'
      elsif (type == 'BlackjackGame')
        game_summary = GameSummary.new
        game_summary.name = 'Blackjack'
        game_summary.link = '/Blackjack/index'
      end
      @games << game_summary
    }
  end
end

File: code/c-sharp/x64/src/OopSolidSample/Controllers/HomeController.cs
using System.Linq;
using System.Web.Mvc;
using OopSolidSample.Domain;
using OopSolidSample.Models.Home;

namespace OopSolidSample.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index()
        {
            var model = new HomeIndexViewModel();
            
            foreach (var gameType in GameBase.AvailableGameTypes)
            {
                var gameModel = new HomeIndexViewModel.GameViewModel();
                if (gameType == typeof(DrawFiveGame))
                {
                    gameModel.Name = "Draw Five";

                    var game = new DrawFiveGame();
                    var highScores = game.GetHighScores();
                    var highestScore = highScores.FirstOrDefault();

                    if (highestScore != null)
                    {
                        gameModel.HighScoreName = highestScore.Name;
                        gameModel.HighScore = highestScore.Score;
                    }
                    gameModel.Link = "/DrawFive";
                }
                else if (gameType == typeof(BlackjackGame))
                {
                    gameModel.Name = "Blackjack";
                    gameModel.Link = "/Blackjack";
                }

                model.Games.Add(gameModel);
            }

            return View(model);
        }
    }
}